.github/workflows/release.yml

```diff
@@ -0,0 +1,96 @@
+        with:
+          registry: docker.elastic.co
+          secret: secret/observability-team/ci/docker-registry/prod
+          url: ${{ secrets.VAULT_ADDR }}
+          roleId: ${{ secrets.VAULT_ROLE_ID }}
+          secretId: ${{ secrets.VAULT_SECRET_ID }}
+
+      - run: make -C .ci env
+
+      - run: make -C .ci dist
+
+      - run: make -C .ci push-docker
+
+      - name: Read AWS vault secrets
+        uses: hashicorp/vault-action@v2.5.0
+        with:
+          url: ${{ secrets.VAULT_ADDR }}
+          method: approle
+          roleId: ${{ secrets.VAULT_ROLE_ID }}
+          secretId: ${{ secrets.VAULT_SECRET_ID }}
+          secrets: |
+            secret/observability-team/ci/service-account/apm-aws-lambda access_key_id | AWS_ACCESS_KEY_ID ;
+            secret/observability-team/ci/service-account/apm-aws-lambda secret_access_key | AWS_SECRET_ACCESS_KEY
+
+      - name: Publish AWS lambda
+        run: make -C .ci publish-in-all-aws-regions create-arn-file
+        env:
+          GH_TOKEN: ${{ github.token }}
+
+      - run: make -C .ci github-release
+        env:
+          GH_TOKEN: ${{ github.token }}
+
+      - name: Read NPM vault secrets
+        uses: hashicorp/vault-action@v2.5.0
+        with:
+          url: ${{ secrets.VAULT_ADDR }}
+          method: approle
+          roleId: ${{ secrets.VAULT_ROLE_ID }}
+          secretId: ${{ secrets.VAULT_SECRET_ID }}
+          secrets: |
+            secret/jenkins-ci/npmjs/elasticmachine token | NPMJS_TOKEN ;
+            totp/code/npmjs-elasticmachine code | TOTP_CODE
+
+      - name: npm publish
+        run: |-
+          echo "//registry.npmjs.org/:_authToken=${{ env.NPMJS_TOKEN }}" > .npmrc
+          npm publish --otp=${{ env.TOTP_CODE }}
+
+      - if: ${{ success() }}
+        name: slack message if success
+        uses: elastic/apm-pipeline-library/.github/actions/slack-message@current
+        with:
+          url: ${{ secrets.VAULT_ADDR }}
+          roleId: ${{ secrets.VAULT_ROLE_ID }}
+          secretId: ${{ secrets.VAULT_SECRET_ID }}
+          channel: "#apm-agent-node"
+          message: |
+            :elasticheart: [${{ github.repository }}] Release *${{ github.ref_name }}* published.
+            Build: (<${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }}|here>)
+
+      - if: ${{ failure() }}
+        name: slack message if failure
+        uses: elastic/apm-pipeline-library/.github/actions/slack-message@current
+        with:
+          url: ${{ secrets.VAULT_ADDR }}
+          roleId: ${{ secrets.VAULT_ROLE_ID }}
+          secretId: ${{ secrets.VAULT_SECRET_ID }}
+          channel: "#apm-agent-node"
+          message: |
+            :broken_heart: [${{ github.repository }}] Release *${{ github.ref_name }}* could not be published.
+            Build: (<${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }}|here>)
```

Comment:
What do you think about utilising https://github.com/elastic/apm-pipeline-library/tree/main/.github/actions/notify-build-status?

Reply:
I could not find how to use a ternary expression in GitHub actions so I can provide a dynamic message instead, but I guess that's not an issue at all since `notify-build-status` uses a template and I can use the message field to highlight the release name.

Let me give a go

Reply:
Yeah, ternaries in actions are tedious.

it would be something like `success() && "yes" || "no"`.

Do you think adding a new param to the action would be beneficial?

Something like `successMessage` and `errorMessage`?

Reply:
No need, we can use the existing implementation. I like how it looks! Thanks for the hint!